core/src/main/java/io/grpc/Metadata.java

```diff
@@ -89,6 +91,27 @@ public String parseAsciiString(String serialized) {
   };
 
   /**
+   * Simple metadata marshaller that encodes strings as is.
+   *
+   * <p>This should be used with ASCII strings that only contain the characters listed in the class
+   * comment of {@link AsciiMarshaller}. Otherwise the output may be considered invalid and
+   * discarded by the transport, or the call may fail.
+   */
+  public static final AsciiMarshaller<List<String>> ASCII_LIST_MARSHALLER =
+      new AsciiMarshaller<List<String>>() {
+
+    @Override
+    public String toAsciiString(List<String> values) {
+      return Joiner.on(',').join(values);
```

Comment:
I looked through RFC 7230 (and 5234) which defines the grammar for header names and values.  Strictly speaking, the presence or absence of commas is header field dependent, and not generally required for any header field.  I am okay deferring this issue until after beta, but it will still need to be fixed.